TALK.md

```diff
@@ -163,8 +162,7 @@ template <typename T>
 dyn_optional<T>& dyn_optional<T>::operator=(dyn_optional<T>&& other) {
     if(this==&other) return *this;
     delete ptr;
-    ptr = nullptr;
-    std::swap(ptr, other.ptr);
```

Comment:
Take or leave this, just thought it could optimise for slide ware

Reply:
Not sure I follow - I think we need this for correctness.

Reply:
Opps, there is an edit missing... It was supposed to say:
```
ptr = std::exchange(other.ptr, nullptr);
```

Reply:
That's a hard-win. Thanks.

Reply:
Applied to main manually.